pkg/portutil/portutil.go

```diff
@@ -73,21 +75,30 @@ func ParseFlagP(s string) ([]gocni.PortMapping, error) {
 	if containerPort == "" {
 		return nil, fmt.Errorf("no port specified: %s", splitBySlash[0])
 	}
-
+	needAutoAssignHostPort := false
 	if hostPort == "" {
-		return nil, fmt.Errorf("automatic host port assignment is not supported yet (FIXME)")
-	}
-
-	startHostPort, endHostPort, err := nat.ParsePortRange(hostPort)
-	if err != nil {
-		return nil, fmt.Errorf("invalid hostPort: %s", hostPort)
+		commandConfig, err := cmd.Flags().GetBool("auto-hostport")
+		if err != nil || !commandConfig {
+			return nil, fmt.Errorf("no host port specified: %s", splitBySlash[0])
```

Comment:
I assume that  `err` msg may be différent then `no host port is specified`. You should return err if not nil before checking the flag value